contracts/src/v0.8/dev/KeeperRegistryDev.sol

```diff
@@ -278,37 +426,22 @@ contract KeeperRegistryDev is
 
   /**
    * @notice updates the configuration of the registry
-   * @param config registry config fields
-   */
-  function setConfig(Config memory config) public onlyOwner {
-    if (config.maxPerformGas < s_storage.maxPerformGas) revert GasLimitCanOnlyIncrease();
-    s_storage = Storage({
-      paymentPremiumPPB: config.paymentPremiumPPB,
-      flatFeeMicroLink: config.flatFeeMicroLink,
-      blockCountPerTurn: config.blockCountPerTurn,
-      checkGasLimit: config.checkGasLimit,
-      stalenessSeconds: config.stalenessSeconds,
-      gasCeilingMultiplier: config.gasCeilingMultiplier,
-      minUpkeepSpend: config.minUpkeepSpend,
-      maxPerformGas: config.maxPerformGas,
-      nonce: s_storage.nonce
-    });
-    s_fallbackGasPrice = config.fallbackGasPrice;
-    s_fallbackLinkPrice = config.fallbackLinkPrice;
-    s_transcoder = config.transcoder;
-    s_registrar = config.registrar;
-    emit ConfigSet(config);
-  }
-
-  /**
-   * @notice update the list of keepers allowed to perform upkeep
-   * @param keepers list of addresses allowed to perform upkeep
-   * @param payees addresses corresponding to keepers who are allowed to
-   * move payments which have been accrued
-   */
-  function setKeepers(address[] calldata keepers, address[] calldata payees) external {
-    // Executed through logic contract
-    _fallback();
+   * @param registryConfig registry configuration fields
+   */
+  function setRegistryConfig(RegistryConfig memory registryConfig) public onlyOwner {
+    if (registryConfig.maxPerformGas < s_config.maxPerformGas) revert GasLimitCanOnlyIncrease();
+    s_config = registryConfig;
+
+    _computeAndStoreConfigDigest(
+      s_signersList,
+      s_transmittersList,
+      s_f,
+      abi.encode(s_config),
+      s_offchainConfigVersion,
+      s_offchainConfig
+    );
+
+    emit RegistryConfigSet(registryConfig);
```

Comment:
let's also (or maybe only?) emit `ConfigSet` and set `onChainConfig = abi.encode(registryConfig)`